agent/engine/errors.go

```diff
@@ -112,6 +112,21 @@ func (CannotGetDockerClientError) ErrorName() string {
 	return "CannotGetDockerclientError"
 }
 
+// TaskCircularDependencyError is the error for task that dependencies can't
+// be resolved
+type TaskCircularDependencyError struct {
```

Comment:
I wonder if `TaskDependencyError` is a better type name. Because, the  `ValidDependencies` method might fail for other reasons as well.